### PR TITLE
fix(ununpack): Increase buffer sizes to prevent overflow

### LINKED ISSUE #1423: Warnings in building ununpack on GCC8
### Logs

```lisp
traverse.c: In function ‘TraverseChild’:
traverse.c:140:9: warning: ‘strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
         strncpy(CI->PartnameNew, UploadFileName, sizeof(CI->PartnameNew) - 1);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ununpack-iso.c: In function ‘ExtractISO’:
ununpack-iso.c:145:123: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4019 and 16304 [-Wformat-truncation=]
     snprintf(Cmd,sizeof(Cmd),"(isoinfo -R -J -i '%s' -x '%s' || isoinfo -R -i '%s' -x '%s' || isoinfo -i '%s' -x '%s') > '%s' 2>/dev/null",TempSource,Line+Len,TempSource,Line+Len,TempSource, Line+Len,Line);
                                                                                                                           ^~                                                                            ~~~~
ununpack-iso.c:145:5: note: ‘snprintf’ output 94 or more bytes (assuming 16474) into a destination of size 16384
     snprintf(Cmd,sizeof(Cmd),"(isoinfo -R -J -i '%s' -x '%s' || isoinfo -R -i '%s' -x '%s' || isoinfo -i '%s' -x '%s') > '%s' 2>/dev/null",TempSource,Line+Len,TempSource,Line+Len,TempSource, Line+Len,Line);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ununpack-disk.c: In function ‘ExtractDisk’:
ununpack-disk.c:477:60: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4078 and 16363 [-Wformat-truncation=]
     snprintf(Cmd,sizeof(Cmd),"icat -f '%s' '%s' '%s' > '%s/%s.deleted.%s' 2>/dev/null",
                                                            ^~
         FStype,TempSource,TempInode,TempDest,TempS,TempInode);
                                              ~~~~~          
ununpack-disk.c:477:5: note: ‘snprintf’ output 44 or more bytes (assuming 20519) into a destination of size 16384
     snprintf(Cmd,sizeof(Cmd),"icat -f '%s' '%s' '%s' > '%s/%s.deleted.%s' 2>/dev/null",
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         FStype,TempSource,TempInode,TempDest,TempS,TempInode);
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ununpack-disk.c:415:60: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4078 and 16363 [-Wformat-truncation=]
     snprintf(Cmd,sizeof(Cmd),"icat -f '%s' '%s' '%s' > '%s/%s' 2>/dev/null",
                                                            ^~
         FStype,TempSource,TempInode,TempDest,TempS);
                                              ~~~~~          
ununpack-disk.c:415:5: note: ‘snprintf’ output 35 or more bytes (assuming 16415) into a destination of size 16384
     snprintf(Cmd,sizeof(Cmd),"icat -f '%s' '%s' '%s' > '%s/%s' 2>/dev/null",
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         FStype,TempSource,TempInode,TempDest,TempS);
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
utils.c: In function ‘IsInflatedFile’:
utils.c:56:3: warning: ‘strncpy’ specified bound 4096 equals destination size [-Wstringop-truncation]
   strncpy(FileNameParent, FileName, sizeof(FileNameParent));
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
utils.c: In function ‘RunCommand’:
utils.c:699:44: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4093 and 12283 [-Wformat-truncation=]
     snprintf(Cmd1,sizeof(Cmd1),"%s %s '%s' %s",
                                            ^~
         Cmd,TempPre,TempFile,TempPost);
                              ~~~~~~~~       
utils.c:699:5: note: ‘snprintf’ output 6 or more bytes (assuming 12291) into a destination of size 12288
     snprintf(Cmd1,sizeof(Cmd1),"%s %s '%s' %s",
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         Cmd,TempPre,TempFile,TempPost);
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
utils.c:694:43: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4094 and 12284 [-Wformat-truncation=]
     snprintf(Cmd1,sizeof(Cmd1),"%s %s '%s/%s' %s",
                                           ^~
         Cmd,TempPre,TempCwd,TempFile,TempPost);
                             ~~~~~~~~       
utils.c:694:5: note: ‘snprintf’ output 7 or more bytes (assuming 16387) into a destination of size 12288
     snprintf(Cmd1,sizeof(Cmd1),"%s %s '%s/%s' %s",
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         Cmd,TempPre,TempCwd,TempFile,TempPost);
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by andi8086**
Fix string buffer sizes to be large enough.

Compiled with gcc 8.3.0 without warnings on ubuntu/bionic.

Signed-off-by: Andreas J. Reichel <andreas.reichel@tngtech.com>

## Description

Prevents a buffer overflow. Closes #1423 

### Changes

Multiplied buffer sizes.

## How to test

Upgrade vagrant vm to ubuntu/bionic and install gcc-8. Go into directory of ununpack agent and build with

make clean
make

### CODE REVIEW

**Comment by GMishx** [comment]
I am still getting following warning on GCC 8.3.0 (`Debian 8.3.0-6`) on Debian 10 (Buster).

```lisp
traverse.c: In function ‘TraverseChild’:
traverse.c:140:9: warning: ‘strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
         strncpy(CI->PartnameNew, UploadFileName, sizeof(CI->PartnameNew) - 1);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
utils.c: In function ‘IsInflatedFile’:
utils.c:56:3: warning: ‘strncpy’ specified bound 4096 equals destination size [-Wstringop-truncation]
   strncpy(FileNameParent, FileName, sizeof(FileNameParent));
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: src/ununpack/agent/traverse.c [hunk 1]
```diff
@@ -137,7 +137,8 @@ void	TraverseChild	(int Index, ContainerInfo *CI, char *NewDir)
       if (CI->TopContainer && UseRepository)
       {
         RemovePostfix(UploadFileName);
-        strncpy(CI->PartnameNew, UploadFileName, sizeof(CI->PartnameNew) - 1);
+        memcpy(CI->PartnameNew, UploadFileName, sizeof(CI->PartnameNew));
+	CI->PartnameNew[sizeof(CI->PartnameNew)-1] = 0;
```
[the remark this answers is not in the document]
Comment (GMishx) [reply src/ununpack/agent/traverse.c]: Just a gentle reminder @andi8086. If you can fix the formatting, we can merge this branch right away.

**Comment by maxhbr** [comment]
@GMishx I have pushed a fix for the formatting

**Comment by GMishx** [comment]
Thank you @maxhbr for providing the fix for formatting.

However, could you please fix the commit message to follow the contributing guideline?

**Comment by maxhbr** [comment]
now the commit matches the conventional changelog requirements.

**Pull request merged**